src/connectedk8s/azext_connectedk8s/_connected_cluster_utils.py

```diff
@@ -0,0 +1,155 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from azure.cli.core import telemetry
+from azure.cli.core.util import sdk_no_wait
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core.azclierror import AzureResponseError, ClientRequestError, AzureInternalError
+from azure.cli.core.azclierror import ValidationError, AzureResponseError, ArgumentUsageError
+from azure.core.exceptions import ResourceNotFoundError
+from msrestazure.azure_exceptions import CloudError
+from msrest.exceptions import ValidationError as MSRestValidationError
+from msrest.exceptions import AuthenticationError, HttpOperationError, TokenExpiredError
+from azext_connectedk8s._client_factory import _resource_providers_client
+from azext_connectedk8s._client_factory import cf_resource_groups
+import azext_connectedk8s._constants as consts
+from azext_connectedk8s._client_factory import _resource_client_factory
+
+
+logger = get_logger(__name__)
+
+
+def check_provider_registrations(cli_ctx):
```

Comment:
Done

Reply:
For 'check_provider_registrations' function, we can do this:
cc_provider_registration = rp_client.get(consts.Connected_Cluster_Provider_Namespace)
cc_registration_state = cc_provider_registration.registration_state

And in validate_location function, we are making the call.
providerDetails = resourceClient.providers.get('Microsoft.Kubernetes')

'providerDetails' should be same as 'cc_provider_registration'. I was asking to make a single ARM call instead of two.
We can use the same client in both functions. Reusing the client from 'validate_location' function would be easier